### PR TITLE
Adding ubi-buildpackless-base builder on integration tests

### PR DESCRIPTION
**Pull request by pacostas**
## Merge after
* https://github.com/paketo-community/ubi-nodejs-extension/pull/48
## Summary

This PR, tests `npm-install` against [ubi buildpackless base](https://github.com/paketo-community/builder-ubi-buildpackless-base) builder in conjuction with [ubi nodejs extention](https://github.com/paketo-community/ubi-nodejs-extension) .

* A new integration test has been added that tests the installation of Node.js addons (native modules) 

## Use Cases

Ensures that ubi-nodejs-extension and UBI images, cooperate with npm-install.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: integration/pre_post_scripts_test.go [hunk 1]
```diff
@@ -52,19 +64,23 @@ func testPrePostScriptRebuild(t *testing.T, context spec.G, it spec.S) {
 		})
 
 		it("does not reach out to the internet", func() {
+
```
Comment (mhdawson) [inline integration/pre_post_scripts_test.go:67]: @pacostas We probably should replace this extra line that is being added.

File: integration/init_test.go [hunk 2]
```diff
@@ -108,9 +125,10 @@ func TestIntegration(t *testing.T) {
 	suite("DevDependenciesDuringBuild", testDevDependenciesDuringBuild)
 	suite("EmptyNodeModules", testEmptyNodeModules)
 	suite("Logging", testLogging)
+	suite("NativeModules", testNativeModules)
 	suite("NoNodeModules", testNoNodeModules)
 	suite("Npmrc", testNpmrc)
-	suite("PackageLockMismatch", testPackageLockMismatch)
+	suite("PackageLockMsismatch", testPackageLockMismatch)
```
Comment (mhdawson) [inline integration/init_test.go:131]: @pacostas I think this was probablyi not on purpose

File: integration/testdata/with_native_modules/node_modules/node-addon-api/.editorconfig [hunk 3]
```diff
@@ -0,0 +1,8 @@
+root = true
+
+[*]
```
Comment (mhdawson) [inline integration/testdata/with_native_modules/node_modules/node-addon-api/.editorconfig:3]: @pacostas I think we can remove this file as well.

**Comment by pacostas** [comment]
rebased.

**Review by thitch97 — DISMISSED** [review]

**Comment by pacostas** [comment]
Tests are failing, probably is due to occam accessing hostports that do not exist, looking at it.

**Comment by pacostas** [comment]
@thitch97 I see that on [ubi-nodejs-extension](https://github.com/paketo-community/ubi-nodejs-extension) the [latest release](https://github.com/paketo-community/ubi-nodejs-extension/releases) is on November 22 and the PR which is prerequisite for the tests to pass, merged on Dec 6. Can you cut a new release on [ubi-nodejs-extension](https://github.com/paketo-community/ubi-nodejs-extension)

**Comment by pacostas** [comment]
@thitch97 Thanks for the new release, can you re-run the tests, probably everything should be working fine.

File: integration/caching_test.go [hunk 4]
```diff
@@ -140,25 +156,25 @@ func testCaching(t *testing.T, context spec.G, it spec.S) {
...
+				extenderBuildStr+"",
+				extenderBuildStr+"    Selected NPM build process: 'npm ci'"))
 			Expect(logs).To(ContainLines(
-				"  Executing launch environment install process",
-				fmt.Sprintf("    Running 'npm ci --unsafe-perm --cache /layers/%s/npm-cache'", strings.ReplaceAll(settings.Buildpack.ID, "/", "_")),
+				extenderBuildStr+"  Executing launch environment install process",
+				fmt.Sprintf(extenderBuildStr+"    Running 'npm ci --unsafe-perm --cache /layers/%s/npm-cache'", strings.ReplaceAll(settings.Buildpack.ID, "/", "_")),
 			))
-			Expect(logs).To(ContainLines(MatchRegexp(`      Completed in (\d+\.\d+|\d{3})`)))
+			Expect(logs).To(ContainLines(MatchRegexp(`\[extender \(build\)\]       Completed in (\d+\.\d+|\d{3})`)))
```
Comment (thitch97) [inline integration/caching_test.go:171]: I think this line is the culprit, it should look more like [this](https://github.com/paketo-buildpacks/npm-install/pull/579/files#diff-22b157d6d07052d616bef4dd70d0304a8161d1753ea5ea7abf55ce0072125221R93-R95).

Comment (pacostas) [reply integration/caching_test.go]: Yeap, thanks, fixed!

**Review by thitch97 — APPROVED** [review]

**Pull request merged**